src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -646,7 +646,15 @@
          Conversion is represented by a single BoundConversion.
          -->
     <Field Name="ConversionGroupOpt" Type="ConversionGroup" Null="allow"/>
-</Node>
+  </Node>
+
+  <!-- A special node for emitting the implicit conversion from an array (of T) to System.ReadOnlySpan<T> -->
+  <Node Name="BoundReadOnlySpanFromArray" Base="BoundExpression">
+    <!-- The type is required an instance of the well-known type System.ReadOnlySpan<T> -->
+    <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
+    <Field Name="Operand" Type="BoundExpression"/>
```

Comment:
>BoundExpression [](start = 32, length = 15)

Can this be a more specific type? (`BoundArrayCreation`)

Reply:
No.  It is intended that the code gen has some flexibility as to what patterns it recognizes and optimizes.

---
In reply to: [281407592](https://github.com/dotnet/roslyn/pull/35463#discussion_r281407592) [](ancestors = 281407592)